amplify/backend/function/getMembersForColony/src/index.js

```diff
@@ -68,209 +67,196 @@ exports.handler = async (event) => {
   }
 
   const {
-    colonyAddress,
+    colonyAddress: address,
     rootHash,
-    domainId = Id.RootDomain,
+    domainId: nativeDomainId = Id.RootDomain,
     sortingMethod = SortingMethod.BY_HIGHEST_REP,
   } = event?.arguments?.input || {};
   const provider = new providers.JsonRpcProvider(rpcURL);
 
+  /*
+   * Validate Colony addresses
+   */
+  let colonyAddress;
+  try {
+    colonyAddress = utils.getAddress(address);
+  } catch (error) {
+    throw new Error(
+      `Colony address "${colonyAddress}" is not valid (after checksum)`,
+    );
+  }
+
   const networkClient = getColonyNetworkClient(network, provider, {
     networkAddress,
     reputationOracleEndpoint,
   });
 
   const colonyClient = await networkClient.getColonyClient(colonyAddress);
-  const { skillId } = await colonyClient.getDomain(domainId || Id.RootDomain);
-  let addressesWithReputation;
+  const { skillId } = await colonyClient.getDomain(nativeDomainId);
+  let contributorAddresses = new Set();
+
   try {
     const { addresses } = await colonyClient.getMembersReputation(skillId);
-    addressesWithReputation = addresses;
+    contributorAddresses = new Set(
+      addresses.map((contributorAddress) => contributorAddress.toLowerCase()),
+    );
   } catch (error) {
-    // @NOTE Not throwing an error here, because its possbile that the colony
+    // @NOTE Not throwing an error here, because it's possbile that the colony
     // doesn't have any reputation yet.
   }
 
-  const watchers = [];
-  const contributors = [];
-
-  /*
-   * Validate Colony addresses
-   */
-  let checksummedAddress;
-  try {
-    checksummedAddress = utils.getAddress(colonyAddress);
-  } catch (error) {
-    throw new Error(
-      `Colony address "${colonyAddress}" is not valid (after checksum)`,
-    );
-  }
-
   // get list of watchers for colony
-  const { data, errors } = await graphqlRequest(
+  const allWatchersResponse = await graphqlRequest(
     getWatchersInColony,
-    { id: checksummedAddress },
+    { id: colonyAddress, domainId: `${colonyAddress}_${nativeDomainId}` },
     graphqlURL,
     apiKey,
   );
 
+  const { data, errors } = allWatchersResponse ?? {};
+
   if (errors || !data) {
     const [error] = errors;
     throw new Error(
       error?.message || 'Could not fetch colony data from DynamoDB',
     );
   }
 
-  // Incase there are members but no reputation on the colony,
-  // we can stop here and return the list of watchers
```

Comment:
Contributors can also have no rep if they have permissions, therefore we can't terminate early here